test/test_dists.jl

```diff
@@ -360,9 +360,10 @@ end
         Q = rand(T, length(x), length(x))
         Q = Q * Q'  # make sure Q is positive-definite
         @test sqmahalanobis(x, y, Q) ≈ dot(x - y, Q * (x - y))
-        @test eltype(sqmahalanobis(x, y, Q)) == T
+        z = oneunit(eltype(Q))
+        @test eltype(sqmahalanobis(x, y, Q)) == typeof(z * (oneunit(eltype(x)) - oneunit(eltype(y))) * z)
```

Comment:
Was this test failing and had to be changed? It seems it should still pass?

Reply:
Thanks for the reviews :+1: 
The tests pass only with  `== T` indeed. There was some mention about reviewing the tests, thought this part might crash. But all is passing as it is. I'll restore the originals